### PR TITLE
[FEATURE]CRUD Users and SchemaUsers

### PR DESCRIPTION
**Pull request by ealexisaraujo**

### CODE REVIEW

File: utils/schema/usersSchema.js [hunk 1]
```diff
@@ -4,14 +4,55 @@ const { config } = require('../../config');
...
+  },
+  password: {
     type: String,
   },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+  modifiedAt: {
+    type: Date,
```
Comment (hyfi06) [inline utils/schema/usersSchema.js:50]: `default: Date.now`

File: utils/schema/usersSchema.js [hunk 2]
```diff
@@ -4,14 +4,55 @@ const { config } = require('../../config');
...
-  LastName: {
+  contactNumber: {
+    type: Number,
+  },
+  username: {
+    type: String,
+    unique: true,
+    lowercase: true,
+    trim: true,
+    default: function () {
```
Comment (hyfi06) [inline utils/schema/usersSchema.js:34]: I think this should be in userServices. You can't control the uniqueness of the username here.

File: utils/schema/usersSchema.js [hunk 3]
```diff
@@ -4,14 +4,55 @@ const { config } = require('../../config');
...
+      const first = this.firstName.split(' ').slice(0, 1).join(' ');
+      const last = this.lastName.split(' ').slice(0, 1).join(' ');
```
Comment (hyfi06) [inline utils/schema/usersSchema.js:36]: The code repeats, crate a function for manipulate text fields. Replaces special characters.

File: utils/schema/usersSchema.js [hunk 4]
```diff
@@ -4,14 +4,55 @@ const { config } = require('../../config');
 const schema = mongoose.Schema;
 
 const UsersSchema = new schema({
-  FirstName: {
+  documentID: {
```
Comment (hyfi06) [inline utils/schema/usersSchema.js:7]: min length 4

File: utils/schema/usersSchema.js [hunk 5]
```diff
@@ -4,14 +4,55 @@ const { config } = require('../../config');
...
   },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+  modifiedAt: {
+    type: Date,
+  },
+  hash: String,
+  salt: String,
```
Comment (hyfi06) [inline utils/schema/usersSchema.js:53]: these fields aren't documented

File: utils/schema/usersSchema.js [hunk 6]
```diff
@@ -4,14 +4,55 @@ const { config } = require('../../config');
...
+    },
+  },
+  password: {
     type: String,
   },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+  modifiedAt: {
```
Comment (hyfi06) [inline utils/schema/usersSchema.js:49]: Standard is `updateAt`. Sorry I didn't know.

File: components/users/usersService.js [hunk 7]
```diff
@@ -13,10 +14,28 @@ class usersService {
     return users || [];
   }
 
+  async createUser({ user }) {
+    const createUserId = await this.mongoDB.create(
+      this.collection,
+      new UserModel(user)
```
Comment (hyfi06) [inline components/users/usersService.js:20]: Add a validation, next milestone. Technical debt

File: utils/schema/usersSchema.js (hunk 3 above)
Comment (hyfi06) [reply utils/schema/usersSchema.js:36]: technical debt: update next milestone

File: utils/schema/usersSchema.js (hunk 1 above)
Comment (ealexisaraujo) [reply utils/schema/usersSchema.js:50]: > `default: Date.now`

This comment is about ? Sorry I dont understand the reference @hyfi06

Comment (hyfi06) [reply utils/schema/usersSchema.js:50]: on create the createAt, and updeteAt have same value.

**Comment by ealexisaraujo** [comment]
I push another commit related to fix issues cc: @hyfi06

**Pull request merged**